From 8f0a6f46ce43a2337ea67e75c44d921bf2700507 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 14 Jan 2015 06:20:55 -0500
Subject: [infiniband] ocrdma: Always resolve destination mac from GRH for UD QPs

Message-id: <22a5e4d3919184cb3ffaf99b8803bd85fab51aa6.1421216208.git.dledford@redhat.com>
Patchwork-id: 102843
O-Subject: [Patch RHEL7 1/2] RDMA/ocrdma: Always resolve destination mac from GRH for UD QPs
Bugzilla: 1167256
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

For user applications that use UD QPs, always resolve destination MAC
from the GRH.  This is to avoid failure due to any garbage value in
the attr->dmac.

Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit e5f0508d43b45d138d426afcaa1e4cce7658cff4)
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_ah.c b/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
index ab29346..a17fc02 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
@@ -96,7 +96,6 @@ struct ib_ah *ocrdma_create_ah(struct ib_pd *ibpd, struct ib_ah_attr *attr)
  struct ocrdma_pd *pd = get_ocrdma_pd(ibpd);
  struct ocrdma_dev *dev = get_ocrdma_dev(ibpd->device);
  union ib_gid sgid;
- u8 zmac[ETH_ALEN];
 
  if (!(attr->ah_flags & IB_AH_GRH))
   return ERR_PTR(-EINVAL);
@@ -118,9 +117,7 @@ struct ib_ah *ocrdma_create_ah(struct ib_pd *ibpd, struct ib_ah_attr *attr)
   goto av_conf_err;
  }
 
- memset(&zmac, 0, ETH_ALEN);
- if (pd->uctx &&
-     memcmp(attr->dmac, &zmac, ETH_ALEN)) {
+ if (pd->uctx) {
   status = rdma_addr_find_dmac_by_grh(&sgid, &attr->grh.dgid,
           attr->dmac, &attr->vlan_id);
   if (status) {
-- 
1.7.1